src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -1990,6 +1993,58 @@ export function getWellFormedFileName(filename: string): string {
 	return filename;
 }
 
+export const SHOW_MODIFICATIONS_SCHEME = 'showModifications';
+export class ShowModificationsAction extends Action implements ITextModelContentProvider {
+	public static ID = 'workbench.files.action.showModifications';
+	public static LABEL = nls.localize('showModifications', "Show modifications");
+
+	private resource: URI;
+
+	constructor(
+		id: string,
+		label: string,
+		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
+		@ITextFileService private textFileService: ITextFileService,
+		@IFileService fileService: IFileService,
+		@IMessageService messageService: IMessageService,
+		@IModeService private modeService: IModeService,
+		@IModelService private modelService: IModelService,
+		@ITextModelService textModelService: ITextModelService
+	) {
+		super(id, label);
+
+		textModelService.registerTextModelContentProvider(SHOW_MODIFICATIONS_SCHEME, this);
+
+		this.enabled = true;
+	}
+
+	public setResource(resource: URI) {
+		this.resource = resource;
+	}
+
+	public run(): TPromise<any> {
+		if (this.resource) {
+			return this.editorService.openEditor({ leftResource: URI.from({ scheme: SHOW_MODIFICATIONS_SCHEME, path: this.resource.fsPath }), rightResource: this.resource, label: nls.localize('Modifications', "Modifications"), options: { pinned: true } });
```

Comment:
I suggest to find a better title for the tab, "Modifications" is very generic. I suggest something similar to the diff editor: 

```typescript
const name = paths.basename(resource.fsPath);
const editorLabel = nls.localize('modifiedLabel', "{0} (on disk) ↔ {1}", name, name);
```